org.jcryptool.core.introduction/src/org/jcryptool/core/introduction/views/AlgorithmInstruction.java

```diff
@@ -260,27 +264,30 @@ public void createPartControl(Composite parent) {
 		gl_content.horizontalSpacing = 0;
 		gl_content.verticalSpacing = 0;
 		content.setLayout(gl_content);
+		content.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
 //		content.setBackground(ColorService.GREEN);
 		
 		scrolledComposite.setContent(content);
 
 
 		initializeScaledImages();
 		
-		
-		// Composite that should grab space on the left.
-		// This should center the slideshow
-		Composite left = new Composite(content, SWT.NONE);
-		GridData gd_left = new GridData(SWT.FILL, SWT.FILL, false, false);
-		left.setLayoutData(gd_left);
-		GridLayout gl_left = new GridLayout();
-		gl_left.marginHeight = 0;
-		gl_left.marginWidth = 0;
-		left.setLayout(gl_left);
+		content.addListener(SWT.Resize, new Listener() {
+			@Override
+			public void handleEvent(Event event) {
+				int[] sizehint = computeSlideshowSizeHint();
+				gridData_cnvs.widthHint = sizehint[0];
+				gridData_cnvs.heightHint = sizehint[1];
+				content.layout(new Control[] {cnvs});
+			}
+		});
```

Comment:
at every resize of the canvas-parent, it's width/heightHint must be computed because it depends on 1) parent size, 2) content aspect ratio

Reply:
If the displayed images change w.r.t. aspect ratio, this calculation and re-layout may be necessary to perform on any image content change as well.